lib/pure/strutils.nim

```diff
@@ -150,23 +150,38 @@ proc isSpaceAscii*(s: string): bool {.noSideEffect, procvar,
   ## characters and there is at least one character in `s`.
   isImpl isSpaceAscii
 
+template isCaseImpl(call) =
+  var hasAtleastOneAlphaChar = false
+  if s.len == 0: return false
+  for c in s:
+    var doCaseCheck = c.isAlphaAscii()
+    if not hasAtleastOneAlphaChar:
+      hasAtleastOneAlphaChar = doCaseCheck
+    if doCaseCheck and (not call(c)): return false
+  if not hasAtleastOneAlphaChar:
```

Comment:
I could have returned just `hasAtleastOneAlphaChar` here, and similar in the other PR. I'll update the PR's tomorrow.